### ISSUE TITLE
An example walk through for new user

### ISSUE THREAD

**Issue by gautamkmr** [kubernetes/sample-controller#54]
I am trying to understand the details of custom controller and planning to write a new one. I could not find a good user doc which talks about the details of file and package. I am looking for something like example other than foo which is being created. The closest I have found is this [article](https://itnext.io/building-an-operator-for-kubernetes-with-the-sample-controller-b4204be9ad56) but this is not quite accurate anymore.

**Comment by SleepyBrett** [comment]
As someone who has written a couple of simple controllers over the last three years I find the addition of the CRD to this controller to be very confusing. Perhaps a sample using only core resource types should be built side by side with this one.

tl;dr this is meant to be a 101, it's a 201.

**Comment by thebsdbox** [comment]
I put this together that details the sample-controller -> https://thebsdbox.co.uk/2019/06/26/Kubernetes-Controllers-Part-One/

The source is all markdown so I can happily donate it if it's any use to the `sample-controller` repo.
